pyproject.toml

```diff
@@ -3,28 +3,74 @@
 # verbose regular expressions by Black. Use [ ] to denote a significant space
 # character.
 
-[tool.black]
-include = '\.pyi?$'
+[tool.ruff]
+# TODO: Some images still use python3.8 in the CI, such as datadog-agent-buildimages/docker_x64 or circleci
+target-version = "py38"
+exclude = [
+    ".git",
+    ".github",
+    ".circleci",
+    "chocolatey",
+    "Dockerfiles",
+    "docs",
+    "google-marketplace",
+    "omnibus",
+    "pkg-config",
+    "releasenotes",
+    "vendor",
+    "venv",
+    "dev",
+]
 line-length = 120
-skip-string-normalization = true
-exclude = '''
-(
-  /(
-    | \.git
-    | docs
-    | releasenotes
-    | releasenotes-installscript
-    | vendor
-    | dev
-  )/
-)
-'''
 
-[tool.isort]
-default_section = 'THIRDPARTY'
-force_grid_wrap = 0
-include_trailing_comma = true
-known_first_party = 'datadog_checks'
-line_length = 120
-multi_line_output = 3
-use_parentheses = true
+[tool.ruff.lint]
+# Rules were ported over from the legacy flake8 settings for parity
+# All the rules can be found here: https://beta.ruff.rs/docs/rules/
+select = [
+    "B",
+    "C",
+    "E",
+    "F",
+    "G",
+    "U",
+    "W",
+    "B003",
+    "B006",
+    "B007",
+]
+ignore = [
+    # From legacy flake8 settings
+    # Ignore:
+    # - black-incompatible options: E203
+    # - bugbear overlap: E722
+    # - style options: W2,W3,W50,E111,E114,E117,E12,E2,E3,E5,E74
+    # - unused argument with underscore: U101
+    # - Unnecessary dict call: C408
+    # - f-string missing prefix FS003
```

Comment:
Nit: you mention E12, U101, and F5003 here, but they are not in the list below. Conversely, C901, B904 are in the list but is not in the comment.

Reply:
Good catch, that's fixed. Thanks